### DISCUSSION TITLE
windows build filament1.9.9 error

### DISCUSSION THREAD

**Discussion by chengjuechao** [google/filament#4870] [Q&A]
I'm compiling from the documentation in Windows.Error while performing step 2 naming(ninja matc resgen cmgen).

> FAILED: third_party/spirv-cross/tnt/CMakeFiles/spirv-cross-msl.dir/__/spirv_msl.cpp.obj
> F:\MICROS~2\VC\Tools\MSVC\1429~1.301\bin\Hostx86\x86\cl.exe  /nologo /TP -DSPIRV_CROSS_EXCEPTIONS_TO_ASSERTIONS -I..\..\third_party\spirv-cross\tnt\.. /DWIN32 /D_WINDOWS /W3 /GR /EHsc /std:c++latest /W0 /Zc:__cplusplus -D_USE_MATH_DEFINES=1 -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE /MD /O2 /Ob2 /DNDEBUG /MT /Zi /showIncludes /Fothird_party\spirv-cross\tnt\CMakeFiles\spirv-cross-msl.dir\__\spirv_msl.cpp.obj /Fdthird_party\spirv-cross\tnt\CMakeFiles\spirv-cross-msl.dir\spirv-cross-msl.pdb /FS -c ..\..\third_party\spirv-cross\spirv_msl.cpp
> cl: 命令行 warning D9025 :正在重写“/W3”(用“/W0”)
> cl: 命令行 warning D9025 :正在重写“/MD”(用“/MT”)
> ..\..\third_party\spirv-cross\spirv_msl.cpp(1327): error C2666: “!=”: 3 个重载有相似的转换
> E:\threeDDemo\filament\third_party\spirv-cross\spirv_common.hpp(416): note: 可为 "bool spirv_cross::TypedID<spirv_cross::TypeFunction>::operator ==(const spirv_cross::TypedID<spirv_cross::TypeNone> &) const" [综合表达式 "!(y == x)"]
> E:\threeDDemo\filament\third_party\spirv-cross\spirv_common.hpp(406): note: 或 "bool spirv_cross::TypedID<spirv_cross::TypeFunction>::operator ==(const spirv_cross::TypedID<spirv_cross::TypeFunction> &) const" [综合表达式 "!(y == x)"]
> ..\..\third_party\spirv-cross\spirv_msl.cpp(1327): note: 或    “内置 C++ operator!=(uint32_t, uint32_t)”
> ..\..\third_party\spirv-cross\spirv_msl.cpp(1327): note: 尝试匹配参数列表“(uint32_t, spirv_cross::FunctionID)”时
> [100/442] Building CXX object third_party\spirv-cross\tnt\CMakeFiles\spirv-cross-core.dir\__\spirv_parser.cpp.obj
> cl: 命令行 warning D9025 :正在重写“/W3”(用“/W0”)
> cl: 命令行 warning D9025 :正在重写“/MD”(用“/MT”)
> ninja: build stopped: subcommand failed.

After performing the first step, the following information appears

> -- The C compiler identification is MSVC 19.29.30137.0
> -- The CXX compiler identification is MSVC 19.29.30137.0
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Check for working C compiler: F:/Microsoft Visual Studio/VC/Tools/MSVC/14.29.30133/bin/Hostx86/x86/cl.exe - skipped
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Check for working CXX compiler: F:/Microsoft Visual Studio/VC/Tools/MSVC/14.29.30133/bin/Hostx86/x86/cl.exe - skipped
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- The ASM compiler identification is MSVC
> -- Found assembler: F:/Microsoft Visual Studio/VC/Tools/MSVC/14.29.30133/bin/Hostx86/x86/cl.exe
> -- DFG LUT size set to 128x128
> -- Found PythonInterp: F:/python-3.8.1/python.exe (found suitable version "3.8.1", minimum required is "3")
> -- The ASM_MASM compiler identification is MSVC
> -- Found assembler: F:/Microsoft Visual Studio/VC/Tools/MSVC/14.29.30133/bin/Hostx86/x86/ml.exe
> -- Configuring done
> -- Generating done
> -- Build files have been written to: E:/threeDDemo/filament/out/cmake-release

How can I solve this problem, please

**Comment by bejado** [comment]
It looks like you're attempting to build for x86, which we don't support on Windows. Please ensure you're working in a `x64 Native Tools Command Prompt for VS 2019` command prompt. You should see something like the following:

```
Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe
```

**Reply by chengjuechao** [reply]
Thank you for your answer, but I use `x64 Native Tools Command Prompt for VS 2019` The same mistake happened.
```
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.7
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

F:\Microsoft Visual Studio>cd E:\threeDDemo\filament\out\cmake-release

F:\Microsoft Visual Studio>e:

E:\threeDDemo\filament\out\cmake-release>cmake ^
More?     -G Ninja ^
More?     -DCMAKE_INSTALL_PREFIX=..\release\filament ^
More?     -DFILAMENT_ENABLE_JAVA=NO ^
More?     -DCMAKE_BUILD_TYPE=Release ^
More? ..\..
-- The C compiler identification is MSVC 19.29.30137.0
-- The CXX compiler identification is MSVC 19.29.30137.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: F:/Microsoft Visual Studio/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: F:/Microsoft Visual Studio/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The ASM compiler identification is MSVC
-- Found assembler: F:/Microsoft Visual Studio/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe
-- DFG LUT size set to 128x128
-- Found PythonInterp: F:/python-3.8.1/python.exe (found suitable version "3.8.1", minimum required is "3")
-- The ASM_MASM compiler identification is MSVC
-- Found assembler: F:/Microsoft Visual Studio/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/ml64.exe
-- Configuring done
-- Generating done
-- Build files have been written to: E:/threeDDemo/filament/out/cmake-release
```

**Reply by bejado** [reply]
I'm not sure what the issue is, but it looks like it's related to the `spirv-cross` library. Version 1.9.9 is over a year old, I'd recommend trying to build a more recent version of Filament.

**Reply by chengjuechao** [reply]
Well, thank you for your help
